Scripts/Source/minai_Arousal.psc

```diff
@@ -529,8 +528,80 @@ Function SetContext(actor akTarget)
     actorRace = StringUtil.Substring(actorRace, 0, cotrIndex)
   EndIf
   aiff.SetActorVariable(akTarget, "race", actorRace)
+
+  string wornEquipments = GetWornEquipments(akTarget)
+  aiff.SetActorVariable(akTarget, "AllWornEquipment", wornEquipments)
+EndFunction
+
+string Function GetWornEquipments(Actor target)
+  ; Encoding format: base form id:esp:slotMasks:keywords:name
+  string wornEquipments = "v1:"
+  int index
+  int slotsChecked
+  slotsChecked += 0x00100000
+  slotsChecked += 0x00200000
+  slotsChecked += 0x80000000
+
+  int currentSlot = 0x01
+  while (currentSlot < 0x80000000)
+    if (Math.LogicalAnd(slotsChecked, currentSlot) != currentSlot) ;only check slots we haven't found anything equipped on already
+      Armor wornArmor = target.GetWornForm(currentSlot) as Armor
+      if (wornArmor)
+        int slotMask = wornArmor.GetSlotMask()
+        string wornArmorName = wornArmor.GetName()
+        ; ensure no pipe in the name
+        wornArmorName = Main.ReplaceString(wornArmorName, ":", "")
```

Comment:
This rather significantly increases the computational costs of this function - It looks pretty cheap otherwise. Can we think of a way to not have to do this on every piece of worn armor? 

This will be getting executed regularly for the PC, as well as all AI NPC's.